From 908e86191bf062711ea44c922e66d27203e90214 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Wed, 8 Jan 2020 10:54:53 +0100
Subject: [PATCH] split declaration and definition of queue variable

Support compilation with -fno-common flag, which is the default for GCC 10.

Fixes: #288
Closes #289
---
 config.c    | 2 ++
 logrotate.h | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/config.c b/config.c
index 0e9a828..f027c7e 100644
--- a/config.c
+++ b/config.c
@@ -28,6 +28,8 @@
 #include "log.h"
 #include "logrotate.h"
 
+struct logInfoHead logs;
+
 #if !defined(GLOB_ABORTED) && defined(GLOB_ABEND)
 #define GLOB_ABORTED GLOB_ABEND
 #endif
diff --git a/logrotate.h b/logrotate.h
index 1c178da..6c1c2e7 100644
--- a/logrotate.h
+++ b/logrotate.h
@@ -89,7 +89,8 @@ struct logInfo {
     TAILQ_ENTRY(logInfo) list;
 };
 
-TAILQ_HEAD(logInfoHead, logInfo) logs;
+TAILQ_HEAD(logInfoHead, logInfo);
+extern struct logInfoHead logs;
 
 extern int numLogs;
 extern int debug;
